homeassistant/components/vizio/media_player.py

```diff
@@ -125,11 +108,11 @@ def __init__(
         self._volume_step = volume_step
         self._current_input = None
         self._available_inputs = None
-        self._device_type = device_type
-        self._supported_commands = SUPPORTED_COMMANDS[device_type]
+        self._device_class = device_class
+        self._supported_commands = SUPPORTED_COMMANDS[device_class]
         self._device = device
         self._max_volume = float(self._device.get_max_volume())
-        self._icon = ICON[device_type]
+        self._icon = ICON[device_class]
```

Comment:
Is the icon still needed after implementing device_class? The frontend should normally handle that when we set a device class, but I'm not sure what the status is for the media_player. How does it look without setting custom icon?

Reply:
Yes it shows the cast icon if I don't set the icon

Reply:
I can remove this from `vizio` and submit a PR for this in the `media_player` integration instead if you think the device class specific icon choices make sense

Reply:
Actually, nevermind. I am not quite sure where the default `media_player` icon is being set

Reply:
It should be part of the frontend repo if it's implemented.